plugins/single_plugins/preserve-output.cpp

```diff
@@ -0,0 +1,314 @@
+    }
+}
+
+nonstd::observer_ptr<last_output_info_t> view_get_data(wayfire_view view)
+{
+    return view->get_data<last_output_info_t>();
+}
+
+bool view_has_data(wayfire_view view)
+{
+    return view->has_data<last_output_info_t>();
+}
+
+void view_erase_data(wayfire_view view)
+{
+    view->erase_data<last_output_info_t>();
+}
+
+struct last_output_info_z_sort
+{
+    inline bool operator ()(nonstd::observer_ptr<wf::view_interface_t> view1,
+        nonstd::observer_ptr<wf::view_interface_t> view2)
+    {
+        return view_get_data(view1)->z_order > view_get_data(view2)->z_order;
+    }
+};
+
+/**
+ * Core last focused output
+ */
+
+class last_focused_output_t : public wf::custom_data_t
+{
+  public:
+    std::string output_identifier = "";
+    std::chrono::time_point<std::chrono::steady_clock> timestamp;
+};
+
+bool core_focused_output_expired()
+{
+    return std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::steady_clock::now() -
+        wf::get_core().get_data_safe<last_focused_output_t>()->timestamp).count() >
+           10000;
+}
+
+void core_store_focused_output(wf::output_t *output)
+{
+    // Store the output as last focused if no other output has been stored as last
+    // focused in the last 10 seconds
+    if ((wf::get_core().get_data_safe<last_focused_output_t>()->output_identifier ==
+         "") || core_focused_output_expired())
+    {
+        LOGD("Setting last focused output to: ", output->to_string());
+        wf::get_core().get_data_safe<last_focused_output_t>()->output_identifier =
+            make_output_identifier(output);
+        wf::get_core().get_data_safe<last_focused_output_t>()->timestamp =
+            std::chrono::steady_clock::now();
+    }
+}
+
+std::string core_get_focused_output()
+{
+    return wf::get_core().get_data_safe<last_focused_output_t>()->output_identifier;
+}
+
+void core_erase_focused_output()
+{
+    wf::get_core().erase_data<last_focused_output_t>();
+}
+
+/**
+ * preserve-output plugin
+ */
+
+class wayfire_preserve_output : public wf::plugin_interface_t
+{
+    std::vector<std::string> outputs_being_removed;
+
+    wf::signal_connection_t output_pre_remove = [=] (wf::signal_data_t *data)
+    {
+        auto signal_data = (wf::output_pre_remove_signal*)data;
+        LOGD("Received pre-remove event: ", signal_data->output->to_string());
+        outputs_being_removed.push_back(signal_data->output->to_string());
+        LOGD("Outputs currently being removed: ");
+        for (auto el : outputs_being_removed)
+        {
+            LOGD("- ", el);
+        }
+
+        if (signal_data->output != output)
+        {
+            // This event is not for this output
+            return;
+        }
+
+        // This output is being destroyed
+        std::string identifier = make_output_identifier(output);
+
+        // Store this output as the focused one
+        if (wf::get_core().get_active_output()->to_string() == output->to_string())
```

Comment:
you can directly compare `get_active_output()` with `output`, no need to compare the strings.